weblate/trans/views/settings.py

```diff
@@ -198,16 +209,27 @@ def remove(request: AuthenticatedHttpRequest, path):
         messages.success(request, gettext("The translation has been removed."))
     elif isinstance(obj, Component):
         parent = obj.category or obj.project
+        obj.to_delete = True
+        obj.save()
```

Comment:
The `save()` here will trigger repository rescan what is not really desired. Better use:

```suggestion
        Component.objects.get(pk=obj.pk).update(to_delete=True)
```

Reply:
Applied this change suggestion.